Cesium.CodeGen/Ir/Expressions/BinaryOperators/ArithmeticBinaryOperatorExpression.cs

```diff
@@ -71,6 +51,64 @@ public override IExpression Lower(IDeclarationScope scope)
         return new ArithmeticBinaryOperatorExpression(left, Operator, right);
     }
 
+    private IExpression LowerPointerArithmetics(IDeclarationScope scope, IExpression left, IExpression right, IType leftType, IType rightType)
+    {
+        if (leftType is PointerType leftPointerType)
+        {
+            if (rightType is PointerType rightPointerType)
+            {
+                if (Operator != BinaryOperator.Subtract)
+                {
+                    throw new CompilationException($"Operator {Operator} is not supported for pointer/pointer operands");
+                }
+
+                var baseSizeLeft = leftPointerType.Base.GetSizeInBytes(scope.ArchitectureSet);
+                var baseSizeRight = rightPointerType.Base.GetSizeInBytes(scope.ArchitectureSet);
+
+                if (baseSizeLeft == null || baseSizeRight == null || baseSizeLeft != baseSizeRight)
+                    throw new CompilationException("Invalid pointer subtraction - pointers are referencing different base types");
+
+                return new ArithmeticBinaryOperatorExpression(
+                    new ArithmeticBinaryOperatorExpression(left, Operator, right),
+                    BinaryOperator.Divide,
+                    new ConstantLiteralExpression(new IntegerConstant(baseSizeLeft.Value))
```

Comment:
Better use `GetSizeInBytesExpression`: it will also manage to use `sizeof` instruction when necessary (i.e. in dynamic arch when the type size is arch-dependent, e.g. a pointer).

Reply:
Done